File: code/01greedy/src/main/java/com/greedy/untilOne/Hyejin.java
package com.greedy.untilOne;

import java.util.Scanner;

public class Hyejin {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("n");
        int n = sc.nextInt();
        System.out.println("k");
        int k = sc.nextInt();
        System.out.println(untilOne(n,k));

        sc.close();
    }

    public static int untilOne(int n, int k){
        int answer = 0;

        while(n!=1){
            answer++;
            n = n%k==0? n/k: n-1;
        }

        return answer;
    }
}

File: src/main/java/com/example/hdreamalgorithm/greedy/coinzero/Hyejin.kt
package com.example.hdreamalgorithm.greedy.coinzero

import java.util.Scanner


//코인
class Hyejin {

    fun main() {
        val sc = Scanner(System.`in`)
        val N = sc.nextInt()
        val K = sc.nextInt()
        val coin = ArrayList<Int>()
        for (i in 0 until N) {
            coin.add(sc.nextInt())
        }
        println(countCoin(K, coin))
    }


    fun countCoin(K: Int, coin: ArrayList<Int>): Int {
        var remainMoney = K
        var totalCoinCount = 0

        for (i in coin.size - 1 downTo 0) {
            if (remainMoney == 0) break
            if (coin[i] > K) continue

            var coinCount = remainMoney / coin[i]

            totalCoinCount += coinCount
            remainMoney -= coinCount * coin[i]

        }
        return totalCoinCount
    }
}

